tests/autologging/test_autologging_utils.py

```diff
@@ -642,3 +645,139 @@ def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with
+    "xgboost": {
+        "package_info": {"pip_release": "xgboost"},
+        "autologging": {"minimum": "0.90", "maximum": "1.2.1"},
+    },
+    "lightgbm": {
+        "package_info": {"pip_release": "lightgbm"},
+        "autologging": {"minimum": "2.3.1", "maximum": "3.1.0"},
+    },
+    "gluon": {
+        "package_info": {"pip_release": "mxnet"},
+        "autologging": {"minimum": "1.5.1", "maximum": "1.7.0.post1"},
+    },
+    "fastai-1.x": {
+        "package_info": {"pip_release": "fastai"},
+        "autologging": {"minimum": "1.0.60", "maximum": "1.0.61"},
+    },
+    "statsmodels": {
+        "package_info": {"pip_release": "statsmodels"},
+        "autologging": {"minimum": "0.11.1", "maximum": "0.12.2"},
+    },
+}
+
+
+@pytest.mark.parametrize(
+    "flavor,module_version_dict,expected_result",
+    [
+        ("fastai", {"fastai": "1.0.60"}, True),
+        ("fastai", {"fastai": "1.0.50"}, False),
+        ("gluon", {"mxnet": "1.6.1"}, True),
+        ("gluon", {"mxnet": "1.5.0"}, False),
+        ("keras", {"keras": "2.2.4"}, True),
+        ("keras", {"keras": "2.2.3"}, False),
+        ("lightgbm", {"lightgbm": "2.3.1"}, True),
+        ("lightgbm", {"lightgbm": "2.3.0"}, False),
+        ("statsmodels", {"statsmodels": "0.11.1"}, True),
+        ("statsmodels", {"statsmodels": "0.11.0"}, False),
+        ("tensorflow", {"tensorflow": "1.15.4"}, True),
+        ("tensorflow", {"tensorflow": "1.15.3"}, False),
+        ("xgboost", {"xgboost": "0.90"}, True),
+        ("xgboost", {"xgboost": "0.89"}, False),
+        ("sklearn", {"sklearn": "0.20.3"}, True),
+        ("sklearn", {"sklearn": "0.20.2"}, False),
+        ("pytorch", {"torch": "1.4.0", "pytorch_lightning": "1.0.5"}, True),
+        ("pytorch", {"torch": "1.3.9", "pytorch_lightning": "1.0.5"}, False),
+        ("pytorch", {"torch": "1.4.0", "pytorch_lightning": "1.0.4"}, False),
+    ],
+)
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_is_autologging_integration_supported(flavor, module_version_dict, expected_result):
+    # Using `ExitStack` to enter a variable list of context, essentially it does something like:
+    # with mock.patch('lib1.__version', 'xxx'),  mock.patch('lib2.__version', 'xxx'), ...
+    with ExitStack() as stack:
+        for module_name in module_version_dict:
+            stack.enter_context(
+                mock.patch(module_name + ".__version__", module_version_dict[module_name])
+            )
+        assert expected_result == is_autologging_integration_supported(flavor)
+
+
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_disable_for_unsupported_versions_warning_sklearn_integration():
+    log_warn_fn_name = "mlflow.utils.autologging_utils._log_warning_for_unsupported_integration"
+
+    with mock.patch("sklearn.__version__", "0.20.3"):
+        AUTOLOGGING_INTEGRATIONS.clear()
```

Comment:
Why do we need to clear `AUTOLOGGING_INTEGRATIONS` each time? I'd expect the warning message to fire even if a preexisting autologging configuration was present. Let me know if I'm missing something here.

Reply:
According to the setup_autologging code, 
if we called  a specific lib autolog first, then call universal autolog, then the latter universal autolog won't change the specific autolog config (this case the specific lib `autolog_fn(**autologging_params)` won't be called so won't fire the warning):
```
            # Only call integration's autolog function with `mlflow.autolog` configs
            # if the integration's autolog function has not already been called by the user.
            # Logic is as follows:
            # - if a previous_config exists, that means either `mlflow.autolog` or
            #   `mlflow.integration.autolog` was called.
            # - if the config contains `CONF_KEY_IS_GLOBALLY_CONFIGURED`, the configuration
            #   was set by `mlflow.autolog`, and so we can safely call `autolog_fn` with
            #   `autologging_params`.
            # - if the config doesn't contain this key, the configuration was set by an
            #   `mlflow.integration.autolog` call, so we should not call `autolog_fn` with
            #   new configs.
```

Reply:
Got it :)! Thanks!